UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -546,6 +549,23 @@ public virtual void InitializeAgent()
         {
         }
 
+        /// <summary>
+        /// Set up the list of ISensors on the Agent. By default, this will select any
+        /// SensorBase's attached to the Agent. To add custom ISensor implementations that are not attached,
+        /// override this method and add them there.
+        /// </summary>
+        public virtual void InitializeSensors()
+        {
+            // TODO deterministic sorting
```

Comment:
Any thoughts on this? Require a name in the sensor (and sensor component)?

Reply:
We need this to be sorted because : 
 - we need all agents who have same sensors to be identifiable
 - we need that even when the user shuffles the sensors, it still works
I don't see another way besides naming them. We could use reflexion to do that (using the type of the sensor if we can ensure there will only ever be one type of sensor per agent...)

Reply:
Done - ISensor now has a GetName() method. Sensors are sorted by that and checked for uniqueness